File: Unidad 5/C#/P5_U5.cs
using System;

class P5_U5
{
    static void Main()
    {
        Console.WriteLine("Ingrese una frase o palabra:");
        string entrada_Usuario = Console.ReadLine();

        int cantidad_Palabras = Contar_Palabras(entrada_Usuario);

        Console.WriteLine("Número de palabras: " + cantidad_Palabras);
    }

    static int Contar_Palabras(string cadena)
    {
        if (string.IsNullOrEmpty(cadena))
        {
            return 0;
        }

        string[] palabras = cadena.Split(new char[] { ' ', '\t', '\n', '\r' }, StringSplitOptions.RemoveEmptyEntries);

        return palabras.Length;
    }
}

File: Unidad 5/JAVA/P5_U5.java
import java.util.Scanner;
public class Main
{
	public static void main(String[] args) {
	  Scanner sc = new Scanner(System.in);
	    
	   System.out.println("Ingrese una frase o una palabra:");
        String entrada_Usuario = sc.nextLine();

        int cantidad_Palabras = contar_Palabras(entrada_Usuario);

        System.out.println("Número de palabras: " + cantidad_Palabras);
    }

    public static int contar_Palabras(String cadena) {
        if (cadena == null || cadena.isEmpty()) {
            return 0;
        }

        String[] palabras = cadena.split("\\s+");

        return palabras.length;
    }
}
